### ISSUE TITLE
🐛 Bug Report: Auth-backend will always fail while resolving the signin profile from catalog because of authentication error

### ISSUE THREAD

**Issue by GuptaNavdeep1983** [backstage/backstage#24569]
### 📜 Description

Backstage version 1.26.3
 
The auth-backend after validating the tokens from upstream provider tries to resolve the identity of the user and find the relevant entity from catalog here,
https://github.com/backstage/backstage/blob/v1.26.3/plugins/auth-backend-module-oauth2-proxy-provider/src/resolvers.ts#L35 and will fail with authentication error as by this time the backstage identity token is not available.

### 👍 Expected behavior

User should be able to resolve the signed in user's profile from catalog after login. If the backstage identity is formed before signing in with catalog backend, the identity can be passed to the backend to avoid the authentication error.

### 👎 Actual Behavior with Screenshots

<img width="623" alt="image" src="https://github.com/backstage/backstage/assets/10557183/4b443659-fb31-4626-9f67-2b1ac26ac36f">

### 👟 Reproduction steps

Login to the oauth2proxy or any other provider with profile transform or signing in resolver using catalog backend should fail as the backstage identity is not formed till then. 

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by benjdlambert** [comment]
@GuptaNavdeep1983 would it be possible to get some more detailed reproduction steps here? Can we get some more info on what is being sent when you get the 401?

**Comment by GuptaNavdeep1983** [comment]
<img width="633" alt="image" src="https://github.com/backstage/backstage/assets/10557183/02626a25-9dcc-4342-a157-ed33f5921dd7">

<img width="658" alt="image" src="https://github.com/backstage/backstage/assets/10557183/da68d6a8-c78e-4b25-98d5-32f20d69fe3c">

<img width="877" alt="image" src="https://github.com/backstage/backstage/assets/10557183/b9903bf3-7b3f-4b50-b5c0-6940bfd1fc08">


Redproduction steps:

I am able to login with same configuration in 1.25.0 release and on the latest release 1.26.3 and 1.26.4, the signin resolver is stuck at the point where its trying to reach to catalog backend with the authentication error. It should be fairly straight forward to reproduce.

**Comment by ljupchokotev** [comment]
Not sure if this is the same issue but we are experiencing something very similar. While migrating to the new backend system, while still using the legacy auth, we noticed that the call: 
```
              const token = await ctx.signInWithCatalogUser({
                annotations: {
                  'ingka/email': profile.email,
                },
              });
```
in our `auth.ts` fails with a `401` status code. The 401 response comes from the catalog itself when to query for a user entity which has that annotation.

We have disabled the default auth policy, but that didn't seem to make any difference.

**Comment by ljupchokotev** [comment]
> I believe you may be hitting this issue if you have set up API auth via the [guide in contrib](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md). With the recent auth changes that guide no longer works and any additions made for it need to be removed.

Not sure if your comments applies to our issue, but we do not have API auth configured. 

In our backend we still make use of `legacyPlugin` to setup auth. That should not break anything in our existing setup as I understand.

**Comment by benjdlambert** [comment]
When you say `In our backend we still make use of legacyPlugin to setup auth` what does this mean. what does your `index.ts` look like?

**Comment by ljupchokotev** [comment]
> When you say `In our backend we still make use of legacyPlugin to setup auth` what does this mean. what does your `index.ts` look like?

We following the migration guide, but for auth we have:
```
backend.add(legacyPlugin('auth', import('./plugins/auth')));
```

**Comment by Rugvip** [comment]
Please make sure you're on 1.27, 1.26 has broken handling of auth with mixed usage of new and legacy plugins in the new system. This was the fix: https://github.com/backstage/backstage/pull/24676

**Comment by ljupchokotev** [comment]
I updated to 1.27 and it seems to fix the issue. The resolver is able to resolve correctly.

There seems to be an issue with the auto-generating an auth key for development, got this error message:
```
[1] Unable to generate legacy token, no legacy keys are configured in 'backend.auth.keys' or 'backend.auth.externalAccess'
```

After adding the following to the config:
```
backend:
  auth:
    keys:
      - secret: test
```

It was able to issue a token and resolve to the correct user entity. 

Are we supposed to also set a backend auth key for development now or is that a regression?

**Comment by camilaibs** [comment]
Hello @ljupchokotev 👋🏻, these keys don't need to be set up for development because the token manager will generate them automatically if `NODE_ENV` is development. 
Could you ping us back if that is not the case?

**Comment by ljupchokotev** [comment]
Hi @camilaibs, that seemed to be the case.

In our `package.json` we have `"dev": "NODE_ENV=development concurrently \"yarn start\" \"yarn start-backend\"",` so that the environment should be set correctly. Before the migration to the new backend system, we were able to run `yarn dev` locally and it would setup the auth keys.

When I upgraded to 1.26.x, I was also able to start the backend without setting up the auth keys. But after the upgrade to 1.27, I had to manually set a key in the config.
